File: Selection_Sort/PHP/mitogh/selection_sort.php
<?php

function selection_sort($unorder = []){
    $n = count($unorder);

    for($j = 0; $j < $n -1; $j++){
        $iMin = $j; 
        for( $i = $j + 1; $i < $n; $i++){
            if($unorder[$i] < $unorder[$iMin]){
                $iMin = $i; 
            }
        }
        if($iMin != $j){
            $t = $unorder[$iMin];
            $unorder[$iMin] = $unorder[$j];
            $unorder[$j] = $t;
        }
    }
    return $unorder;
}

File: Selection_Sort/Lua/Yonaba/selection_sort.lua
-- Selection sort algorithm
-- See: http://en.wikipedia.org/wiki/Selection_sort

-- list: a list to be sorted, in-place
-- comp: (optional) a comparison function
--  defaults to function(a, b) return a < b end
-- returns: the passed-in list, sorted
return function(list, comp)
	comp = comp or function(a, b) return a < b end
	local n = #list
	for i = 1, n do
		local iMin = i
		for j = iMin+1, n do
			if not comp(list[iMin],list[j]) then
				iMin = j
			end
		end
		if iMin~=i then
			list[i], list[iMin] = list[iMin], list[i]
		end
	end
	return list
end
